## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 466008 --additional-package nfs-utils.passthru.tests --additional-package pkgsMusl.nfs-utils`
Commit: `ce52a617c66f721a773e366cc634f64f2bd9037c`

---
### `x86_64-linux`
<details>
  <summary>:x: 20 packages failed to build:</summary>
  <ul>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client (ceph-client.client, ceph-client.dev, ceph-client.doc, ceph-client.lib, ceph-client.man, ceph-dev.client, ceph.client)</li>
    <li>ceph-csi</li>
    <li>ceph.dev (ceph-dev.dev)</li>
    <li>ceph.doc (ceph-dev.doc)</li>
    <li>libceph (ceph.lib, libceph.client, libceph.dev, libceph.doc, libceph.lib, libceph.man)</li>
    <li>ceph.man (ceph-dev.man)</li>
    <li>multipass</li>
    <li>nfs-utils.passthru.tests.nfs3-simple</li>
    <li>pkgsMusl.nfs-utils</li>
    <li>pkgsMusl.nfs-utils.dev</li>
    <li>pkgsMusl.nfs-utils.lib</li>
    <li>pkgsMusl.nfs-utils.man</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.doc</li>
    <li>qemu_full.ga</li>
    <li>sambaFull (samba4Full)</li>
    <li>sambaFull.dev (samba4Full.dev)</li>
    <li>sambaFull.man (samba4Full.man)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 87 packages built:</summary>
  <ul>
    <li>appvm</li>
    <li>autofs5</li>
    <li>booster</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>calamares.debug</li>
    <li>collectd</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>facter</li>
    <li>fdroidserver</li>
    <li>fdroidserver.dist</li>
    <li>freeipa</li>
    <li>gnome-boxes</li>
    <li>kdePackages.kpmcore</li>
    <li>kdePackages.kpmcore.debug</li>
    <li>kdePackages.kpmcore.dev</li>
    <li>kdePackages.kpmcore.devtools</li>
    <li>kdePackages.partitionmanager</li>
    <li>kdePackages.partitionmanager.debug</li>
    <li>kdePackages.partitionmanager.dev</li>
    <li>kdePackages.partitionmanager.devtools</li>
    <li>libguestfs (python313Packages.guestfs)</li>
    <li>libguestfs.guestfsd (python313Packages.guestfs.guestfsd)</li>
    <li>librenms</li>
    <li>libvirt</li>
    <li>libvirt-dbus</li>
    <li>libvirt-dbus.man</li>
    <li>libvirt-glib</li>
    <li>libvirt-glib.dev</li>
    <li>libvirt-glib.devdoc</li>
    <li>libvmi</li>
    <li>libvmi.dev</li>
    <li>libvmi.lib</li>
    <li>lockbook</li>
    <li>mgmt</li>
    <li>minikube</li>
    <li>nagiosPlugins.check_zfs</li>
    <li>nfs-ganesha</li>
    <li>nfs-ganesha.man</li>
    <li>nfs-ganesha.tools</li>
    <li>nfs-utils</li>
    <li>nfs-utils.dev</li>
    <li>nfs-utils.lib</li>
    <li>nfs-utils.man</li>
    <li>nfs-utils.passthru.tests.nfs4-kerberos</li>
    <li>nfs-utils.passthru.tests.nfs4-simple</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>perl538Packages.SysVirt</li>
    <li>perl538Packages.SysVirt.devdoc</li>
    <li>perlPackages.SysVirt (perl540Packages.SysVirt)</li>
    <li>perlPackages.SysVirt.devdoc (perl540Packages.SysVirt.devdoc)</li>
    <li>podman-bootc</li>
    <li>prometheus-libvirt-exporter</li>
    <li>python312Packages.guestfs</li>
    <li>python312Packages.guestfs.guestfsd</li>
    <li>python312Packages.libvirt</li>
    <li>python312Packages.libvirt.dist</li>
    <li>python312Packages.py-libzfs</li>
    <li>python312Packages.py-libzfs.dist</li>
    <li>python312Packages.sss</li>
    <li>python313Packages.libvirt</li>
    <li>python313Packages.libvirt.dist</li>
    <li>python313Packages.sss</li>
    <li>rubyPackages.ruby-libvirt (rubyPackages_3_3.ruby-libvirt)</li>
    <li>rubyPackages_3_4.ruby-libvirt</li>
    <li>rubyPackages_3_5.ruby-libvirt</li>
    <li>sanoid</li>
    <li>sssd</li>
    <li>tzpfms</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-top</li>
    <li>virt-v2v</li>
    <li>virt-viewer</li>
    <li>virtnbdbackup</li>
    <li>virtnbdbackup.dist</li>
    <li>zfs (zfs_2_3)</li>
    <li>zfs.dev (zfs_2_3.dev)</li>
    <li>zfs_unstable</li>
    <li>zfs_unstable.dev</li>
    <li>zfstools</li>
    <li>zpool-auto-expand-partitions</li>
    <li>zxfer</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>nfs-utils.passthru.tests.nfs3-simple</summary>
<pre>client1: must fail: time flock -n -s /data/lock true
client1 # 
client1 # real  0m0.041s
client1 # user  0m0.004s
client1 # sys   0m0.000s
!!! Test &quot;locks survive server reboot&quot; failed with error: &quot;command `time flock -n -s /data/lock true` unexpectedly succeeded&quot;
!!! Traceback (most recent call last):
!!!   File &quot;&lt;string&gt;&quot;, line 43, in &lt;module&gt;
!!!     client1.fail(&quot;time flock -n -s /data/lock true&quot;)
!!! 
!!! RequestedAssertionFailed: command `time flock -n -s /data/lock true` unexpectedly succeeded
cleanup
kill machine (pid 120)
qemu-system-x86_64: terminating on signal 15 from pid 8 (/nix/store/3lll9y925zz9393sa59h653xik66srjb-python3-3.13.9/bin/python3.13)
kill machine (pid 67)
qemu-system-x86_64: terminating on signal 15 from pid 8 (/nix/store/3lll9y925zz9393sa59h653xik66srjb-python3-3.13.9/bin/python3.13)
kill machine (pid 150)
qemu-system-x86_64: terminating on signal 15 from pid 8 (/nix/store/3lll9y925zz9393sa59h653xik66srjb-python3-3.13.9/bin/python3.13)
kill vlan (pid 9)
(finished: cleanup, in 0.01 seconds)</pre>
</details>
<details>
<summary>pkgsMusl.nfs-utils</summary>
<pre>checking for memset... yes
checking for mkdir... yes
checking for pathconf... yes
checking for ppoll... yes
checking for realpath... yes
checking for rmdir... yes
checking for select... yes
checking for socket... yes
checking for strcasecmp... yes
checking for strchr... yes
checking for strdup... yes
checking for strerror... yes
checking for strrchr... yes
checking for strtol... yes
checking for strtoul... yes
checking for sigprocmask... yes
checking for name_to_handle_at... yes
checking for getrpcbynumber... no
checking for getrpcbynumber_r... no
configure: error: Neither getrpcbynumber_r nor getrpcbynumber are available</pre>
</details>
</details>

--- a/pkgs/by-name/nf/nfs-utils/package.nix
+++ b/pkgs/by-name/nf/nfs-utils/package.nix
@@ -181,6 +181,8 @@ stdenv.mkDerivation rec {
     nfs4-kerberos = nixosTests.nfs4.kerberos;
   };
 
+  passthru.updateScript = ./update.sh;
+
   meta = with lib; {
     description = "Linux user-space NFS utilities";
 

--- a/pkgs/by-name/nf/nfs-utils/package.nix
+++ b/pkgs/by-name/nf/nfs-utils/package.nix
@@ -195,6 +195,6 @@ stdenv.mkDerivation rec {
     homepage = "https://linux-nfs.org/";
     license = licenses.gpl2Plus;
     platforms = platforms.linux;
-    maintainers = [ ];
+    maintainers = [ lib.maintainers.dotlambda ];
   };
 }

--- a/pkgs/by-name/nf/nfs-utils/update.sh
+++ b/pkgs/by-name/nf/nfs-utils/update.sh
@@ -1,0 +1,10 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p coreutils gnugrep common-updater-scripts nix-update
+
+# tags look like nfs-utils-2-8-4
+version=$(list-git-tags --url=git://git.linux-nfs.org/projects/steved/nfs-utils.git | grep -oP '^nfs-utils-\d+-\d+-\d+$' | sort -rV | head -1)
+# remove nfs-utils- prefix
+version="${version#nfs-utils-}"
+# replace - with .
+version="${version//-/.}"
+nix-update --version="$version" nfs-utils

